src/main/java/org/terasology/spawning/SpawningAuthoritySystem.java

```diff
@@ -0,0 +1,171 @@
+                    logger.info("Could not find all required items in inventory");
+                    return false;
+                }
+
+                for (int slotNumber = 0; slotNumber < numberOfItems; slotNumber++) {
+                    if(!removeNeededItem(itemsHashMap, requiredSlots.get(slotNumber), player)) {
+                        //could not decrement the required number from inventory
+                        return false;
+                    }
+                }
+
+                logger.info("Found all items required for spawning, creating your Oreon");
+                return true;
+            }
+
+            logger.info("No inventory to source material from, cannot spawn");
+            return false;
+        }
+
+        logger.info("No item required to spawn, spawning your Oreon");
+        return true;
+    }
+
+    public ArrayList<Integer> getSlotsForRequiredItems(HashMap<String, Integer> items, EntityRef player) {
+        ArrayList<Integer> requiredSlots = new ArrayList<>();
+
+        int inventorySize = inventoryManager.getNumSlots(player);
+
+        for(int slotNumber = 0; slotNumber <= inventorySize; slotNumber++) {
+            EntityRef inventorySlot = inventoryManager.getItemInSlot(player, slotNumber);
+
+            BlockItemComponent blockItemComponent = inventorySlot.getComponent(BlockItemComponent.class);
+            if (blockItemComponent != null) {
+                String blockFamily = blockItemComponent.blockFamily.toString();
+                //if this item is required
+                if (items.containsKey(blockFamily)) {
+                    requiredSlots.add(slotNumber);
+                }
+            }
+        }
+
+        return requiredSlots;
+    }
+
+    public boolean removeNeededItem(HashMap<String, Integer> items, int slotNumber, EntityRef player) {
+        EntityRef inventorySlot = inventoryManager.getItemInSlot(player, slotNumber);
+        BlockItemComponent blockItemComponent = inventorySlot.getComponent(BlockItemComponent.class);
+        String blockFamily = blockItemComponent.blockFamily.toString();
+
+        logger.info("This Oreon has an item demand for spawning: " + blockFamily);
+        logger.info("Found the item needed to spawn stuff! Decrementing by {}, then spawning", items.get(blockFamily));
+
+        EntityRef result = inventoryManager.removeItem(player, player, inventorySlot, false, items.get(blockFamily));
+        if(result == null) {
+            logger.info("Could not decrement the required amount from inventory, not spawning");
+            return false;
+        }
+
+        //successfully removed the required number of needed item
+        return true;
+    }
+
+    public void assignRandomAttributes(EntityRef oreon) {
+        OreonAttributeComponent oreonAttributes = new OreonAttributeComponent();
+
+        MersenneRandom random = new MersenneRandom();
+
+        if(prefabToSpawn.getName().equals("Oreons:OreonBuilder")) {
```

Comment:
Are you sure getName() is non-null?  Here's a case where using the constant on the left-hand-side is a good idea to avoid an explicit null check.

Second, "Oreons:OreonBuilder" should be declared as a constant so it can easily be changed later.